drivers/media/i2c/ov9281.c

```diff
@@ -613,6 +618,34 @@ static int ov9281_enable_test_pattern(struct ov9281 *ov9281, u32 pattern)
 				OV9281_REG_VALUE_08BIT, val);
 }
 
+static int ov9281_set_ctrl_hflip(struct ov9281 *ov9281, int value)
+{
+	u32 current_val;
+	int ret = ov9281_read_reg(ov9281->client, OV9281_REG_TIMING_FORMAT_2,
+			      OV9281_REG_VALUE_08BIT, &current_val);
+	if (!ret)
+	{
+		current_val ^= (-value ^ current_val) & (1 << 2);		
```

Comment:
I don't really even want to think through this one.

negative signed value XOR with an unsigned value, and then logical AND, all XOR with the number you first thought of.

Simple logic of always clearing the bit (use a ```#define```), and then conditionally setting it please (reuse same ```#define```).

Reply:
Sorry for this one, it is a bit tricky. I saw you are using the BIT() macro - where does it come from and is there more help like this? I added macros to SET and UNSET a bit (not sure if this already exists), because this a typical task in these drivers I would say.